sdk/tests/conformance2/textures/misc/copy-texture-image-webgl-specific.html

```diff
@@ -0,0 +1,149 @@
+    gl.texImage3D(gl.TEXTURE_3D, 0, gl.RGBA, width, height, depth, 0, gl.RGBA, gl.UNSIGNED_BYTE, uint8);
+
+    var fbo = gl.createFramebuffer();
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+    gl.framebufferTextureLayer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, texture, 0, layer);
+    if (gl.checkFramebufferStatus(gl.FRAMEBUFFER) == gl.FRAMEBUFFER_COMPLETE) {
+        gl.copyTexSubImage3D(gl.TEXTURE_3D, 0, 0, 0, layer, 0, 0, width, height);
+            wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "gl.INVALID_OPRATION is generated");
+    } else {
+        testFailed("framebuffer not complete");
+    }
+
+    gl.bindFramebuffer(gl.FRAMEBUFFER, null);
+    gl.deleteFramebuffer(fbo);
+    gl.deleteTexture(texture);
+    testPassed("gl.INVALID_OPRATION is generated if the source and destination textures are the same.");
+};
+
+function copytexsubimage3d_valid_operation_diff_level() {
+    debug("");
+    debug("Testing copytexsubimage3d_valid_operation_diff_level");
+    var texture = gl.createTexture();
+    gl.bindTexture(gl.TEXTURE_3D, texture);
+    var uint8 = new Uint8Array(32);
+    var layer1 = 0;
+    var layer2 = 1;
+    var width = 2;
+    var height = 2;
+    var depth = 2;
+    gl.texImage3D(gl.TEXTURE_3D, 0, gl.RGBA, width, height, depth, 0, gl.RGBA, gl.UNSIGNED_BYTE, uint8);
+
+    var fbo = gl.createFramebuffer();
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+    gl.framebufferTextureLayer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, texture, 0, layer1);
+    if (gl.checkFramebufferStatus(gl.FRAMEBUFFER) == gl.FRAMEBUFFER_COMPLETE) {
+        gl.copyTexSubImage3D(gl.TEXTURE_3D, 0, 0, 0, layer2, 0, 0, width, height);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texImage3D should succeed.");
+    } else {
+        testFailed("framebuffer not complete");
+    }
+
+    gl.bindFramebuffer(gl.FRAMEBUFFER, null);
+    gl.deleteFramebuffer(fbo);
+    gl.deleteTexture(texture);
+    testPassed("gl.NO_ERROR is generated if the layer of source and destination textures are different.");
+};
+
+function copytexsubimage3d_valid_operation_diff_layer() {
+    debug("");
+    debug("Testing copytexsubimage3d_valid_operation_diff_layer");
+    var texture = gl.createTexture();
+    gl.bindTexture(gl.TEXTURE_3D, texture);
+    var uint8 = new Uint8Array(32);
+    var layer1 = 0;
+    var layer2 = 1;
+    var width = 2;
+    var height = 2;
+    var depth = 2;
+    gl.texImage3D(gl.TEXTURE_3D, 0, gl.RGBA, width, height, depth, 0, gl.RGBA, gl.UNSIGNED_BYTE, uint8);
+
+    var fbo = gl.createFramebuffer();
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+    gl.framebufferTextureLayer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, texture, 0, layer1);
+    if (gl.checkFramebufferStatus(gl.FRAMEBUFFER) == gl.FRAMEBUFFER_COMPLETE) {
+        gl.copyTexSubImage3D(gl.TEXTURE_3D, 0, 0, 0, layer2, 0, 0, width, height);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texImage3D should succeed.");
+    } else {
+        testFailed("framebuffer not complete");
```

Comment:
The original testPassed() is not necessary, but I suggest Yizhou to keep the deinit code, no matter it fails or successes. 
So she deleted the original testPassed() after deinit. 
PTAL. @kenrussell and @zhenyao.